contrib/devtools/update-clang-hashes.sh

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env bash
+
+export LC_ALL=C
+set -o pipefail
+
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+CLANG_PACKAGE="$SCRIPT_DIR/../../depends/packages/native_clang.mk"
+LIBCXX_PACKAGE="$SCRIPT_DIR/../../depends/packages/libcxx.mk"
+
+CLANG_MAJOR_VERSION=$( cat $CLANG_PACKAGE | grep -oP "_major_version=\K.*" )
+CLANG_VERSION=$( cat $CLANG_PACKAGE | grep -v _major_version | grep -oP "_version=\K.*" )
+LIBCXX_MSYS2_VERSION=$( cat $LIBCXX_PACKAGE | grep -oP "_msys2_version=\K.*" )
+
+update_clang_hash() {
+    url="https://github.com/llvm/llvm-project/releases/download/llvmorg-$CLANG_VERSION/clang+llvm-$CLANG_VERSION-$1.tar.xz"
+    echo "Fetching $url"
+    hash=$( curl -fL $url | sha256sum | awk '{print $1}' )
+    retVal=$?
+    if [ $retVal -ne 0 ]; then
+        if [ $retVal -eq 22 ]; then
+            echo
+            echo "The LLVM project has not published a $CLANG_VERSION build for $1."
+            echo "You will need to manually fix the Makefile to use a different version."
+            echo
+        fi
```

Comment:
This appears to do nothing if `$retVal` is neither 0 nor 22. Is that intended?

Reply:
Correct; `curl` already prints out an error message, this is just indicating how to resolve the most common one.